ledger-tool/src/output.rs

```diff
@@ -0,0 +1,56 @@
+use {
+    serde::Serialize,
+    solana_cli_output::{QuietDisplay, VerboseDisplay},
+    std::fmt::{Display, Formatter, Result},
+};
+
+#[derive(Serialize, Debug, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct SlotInfo {
+    pub total: usize,
+    pub first: Option<u64>,
+    pub last: Option<u64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub num_after_last_root: Option<usize>
+}
+
+impl Display for SlotInfo {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
+        writeln!(f, "Total: {}", &self.total)?;
+        if let Some(first) = &self.first {
+            writeln!(f, "First: {}", first)?;
+        }
+        if let Some(last) = &self.last {
+            writeln!(f, "Last: {}", last)?;
+        }
+        if let Some(num_after_last_root) = &self.num_after_last_root {
+            writeln!(f, "From Last: {}", num_after_last_root)?;
+        }
+        Ok(())
+    }
+}
+
+#[derive(Serialize, Debug, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct SlotBounds {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub slots: Option<SlotInfo>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub roots: Option<SlotInfo>,
```

Comment:
To match the suggestion made elsewhere to make the first/last fields options, these should no longer be options
```suggestion
    pub slots: SlotInfo,
    pub roots: SlotInfo,
```

Reply:
About this, is the logic to print info there's also an `all` parameter coming from `matches` took from command line options. This parameter has the following purpose:
```
if all {
    println!("Non-empty slots: {:?}", slots);
}
```
Is it ok to add another parameter (a `bool` for instance) inside `SlotBounds` in order to accomplish this logic?

Reply:
Yes, you can add a parameter tagged with either `#[serde(skip_serializing)]` or `#[serde(skip_serializing_if = "Option::is_none")]`, depending on what kind of field you need, to pass data to use in the Display implementation, without returning that data in the JSON map.